### PR TITLE
Migrate to OpenAPI 3.1

### PR DESCRIPTION
**Pull request by justnero**
## Summary

A separate bundle for SDK generation is a temporary solution that would be here until PHP SDK generator gets better OAS 3.1 support.

Changes:
- Replaced `nullable` with type arrays
- Replaced `x-webhooks` with `webhooks`
- Added decorator to build OpenAPI 3.0.3 version for SDK generation

<details>
<summary>Replace `nullable` with type arrays</summary>

Search:
```regex
^(?<space> *)(?:nullable: true(?<between1>(?:\n\k<space>[^\n]*)*?)\n\k<space>type: (?<type1>\w+)|type: (?<type2>\w+)(?<between2>(?:\n\k<space>[^\n]*)*?)\n\k<space>nullable: true)
```
Replace:
```regex
$<space>type:\n$<space>  - '$<type1>$<type2>'\n$<space>  - 'null'$<between1>$<between2>
```
VSCode friendly version: `$1type:\n$1  - '$3$4'\n$1  - 'null'$2$5`

</details>

## Links
https://github.com/Rebilly/rebilly/issues/5304

## Checklist

- [x] Writing style
- [x] API design standards

### CODE REVIEW

File: openapi/components/requestBodies/PatchFee.yaml [hunk 1]
```diff
@@ -15,16 +15,18 @@ properties:
...
     description: |-
       Fee settlement settings.
       This value overrides the gateway account financial settings of the transaction.
-    type: object
-    nullable: true
+    type:
+      - 'object'
+      - 'null'
     allOf:
       - $ref: ../schemas/SettlementSettings.yaml
```
Comment (adamaltman) [inline openapi/components/requestBodies/PatchFee.yaml:31]: This is nonsense though.

I think your replacement logic works for type string, null, but not for type object.

File: openapi/components/requestBodies/storefront/SetupPaymentInstrumentRequest.yaml [hunk 2]
```diff
@@ -22,19 +22,21 @@ content:
             If a billing address is not supplied the address associated with the payment instrument is used.
             If no address is associated with the payment instrument,
             the customer's address is used.
-          nullable: true
-          type: object
+          type:
```
Comment (adamaltman) [inline openapi/components/requestBodies/storefront/SetupPaymentInstrumentRequest.yaml:25]: I can guess that this is wrong based on the `allOf`.  I'll follow up with a comment if I'm right or wrong.

Comment (adamaltman) [reply openapi/components/requestBodies/storefront/SetupPaymentInstrumentRequest.yaml:25]: 🎉 I am right.

**Review by adamaltman — APPROVED** [review]

File: plugins/products-bundler/bundler.js [hunk 3]
```diff
@@ -49,9 +49,9 @@ const decorators = {
               definitionRoot['servers'] = productMapping['servers'];
             }
             definitionRoot['paths'] = getNewPaths(definitionRoot['paths'], ctx, tagsNamesToInclude, availableMethods, includedXProducts);
-            definitionRoot['x-webhooks'] = getNewPaths(definitionRoot['x-webhooks'], ctx, tagsNamesToInclude, ['post'], includedXProducts);
-            if (Object.keys(definitionRoot['x-webhooks']).length === 0) {
-              delete definitionRoot['x-webhooks'];
+            definitionRoot['webhooks'] = getNewPaths(definitionRoot['webhooks'], ctx, tagsNamesToInclude, ['post'], includedXProducts);
```
Comment (adamaltman) [inline plugins/products-bundler/bundler.js:52]: We don't use this bundler any longer. I was going to delete it.

Comment (justnero) [reply plugins/products-bundler/bundler.js:52]: I'll remove it separately then

**Review by wmdmgv — APPROVED** [review]

**Pull request merged**